fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,125 @@
+#include "fdbserver/MockPeekCursor.h"
+#include "fdbserver/MockLogSystem.h"
+#include "fdbclient/StorageServerInterface.h"
+#include "fdbserver/IKeyValueStore.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "fdbserver/RecoveryState.h"
+#include "fdbserver/ptxn/test/Driver.h"
+
+#include "flow/UnitTest.h"
+
+#include "flow/actorcompiler.h" // has to be last include
+
+namespace {
+    StringRef fileStoragePrefix = LiteralStringRef("storage-");
+
+    // Copyed from fdbserver/worker.actor.cpp for convenience.
+    std::string filenameFromId(KeyValueStoreType storeType, std::string folder, std::string prefix, UID id) {
+        if (storeType == KeyValueStoreType::SSD_BTREE_V1)
+            return joinPath(folder, prefix + id.toString() + ".fdb");
+        else if (storeType == KeyValueStoreType::SSD_BTREE_V2)
+            return joinPath(folder, prefix + id.toString() + ".sqlite");
+        else if (storeType == KeyValueStoreType::MEMORY || storeType == KeyValueStoreType::MEMORY_RADIXTREE)
+            return joinPath(folder, prefix + id.toString() + "-");
+        else if (storeType == KeyValueStoreType::SSD_REDWOOD_V1)
+            return joinPath(folder, prefix + id.toString() + ".redwood");
+        else if (storeType == KeyValueStoreType::SSD_ROCKSDB_V1)
+            return joinPath(folder, prefix + id.toString() + ".rocksdb");
+
+        UNREACHABLE();
+    }
+}
+
+TEST_CASE("fdbserver/ptxn/test/storageserver") {
+    using namespace ptxn;
+
+    UID uid = nondeterministicRandom()->randomUniqueID();
+    state StorageServerInterface recruited(uid);
+    recruited.locality = LocalityData();
+    recruited.initEndpoints();
+
+    KeyValueStoreType storeType = KeyValueStoreType::SSD_BTREE_V2;
+    state std::string folder = ".";
+    std::string fileName = filenameFromId(storeType, folder, fileStoragePrefix.toString(), recruited.id());
+    std::cout << "new SS file name: " << fileName << std::endl;
+    deleteFile(fileName);
+    state IKeyValueStore* data = openKVStore(storeType, fileName, recruited.id(), 0);
+
+    state Tag tag = Tag(1, 1);
+
+    state ReplyPromise<InitializeStorageReply> storageReady;
+
+    // Initialize dbInfo with mocked cursor and mocked loy system.
+    Arena arena;
+
+    MutationRef mutation(arena,
+                         MutationRef::SetValue,
+                         StringRef(std::string("Hello")),
+                         StringRef(std::string("World")));
```

Comment:
We typically use `LiteralStringRef("Hello")` or use the shorthand version `"Hello"_sf`.

Reply:
Fixed.